core/src/main/scala/besom/internal/ResourceOps.scala

```diff
@@ -132,13 +143,25 @@ class ResourceOps(using ctx: Context, mdc: BesomMDC[Label]):
     for
       maybeProviderId <- maybeProviderIdResult
       req             <- buildInvokeRequest(invokeArgs.serialized, maybeProviderId, version)
-      _               <- log.debug(s"Invoke RPC prepared, req=${pprint(req)}")
+      _               <- log.debug(s"Invoke RPC prepared, req:\n${pprint(req)}")
       res             <- ctx.monitor.invoke(req)
-      _               <- log.debug(s"Invoke RPC executed, res=${pprint(res)}")
+      _               <- log.debug(s"Invoke RPC executed, res:\n${pprint(res)}")
       parsed          <- parseInvokeResponse(tok, res)
     yield parsed
   end executeInvoke
 
+  private[internal] def resolveMode(options: ResourceOptions): Result[Mode] =
+    // this order is then repeated in registerReadOrGetResource
+    options.hasURN.zip(options.hasImportId).flatMap { case (hasUrn, hasImportId) =>
```

Comment:
Nit: I'd move this internal logic to the companion object of the Mode and use pat-mat instead of if for readability because it allow for a sort of "table of allowed states"

Reply:
can't be patmat, this is async